guidance/library/_json.py

```diff
@@ -34,7 +35,47 @@ def _to_compact_json(target: Any) -> str:
     return json_dumps(target, separators=(",", ":"))
 
 
-_DEFS_KEYS = ["$defs", "definitions"]
+class Keyword(str, Enum):
+    ANYOF = "anyOf"
+    ALLOF = "allOf"
+    REF = "$ref"
+    CONST = "const"
+    ENUM = "enum"
+    TYPE = "type"
+
+
+KEYS = {member.value for member in Keyword}
+
+DEFS_KEYS = {"$defs", "definitions"}
+
+IGNORED_KEYS = {
+    "$schema",
+    "$id",
+    "$comment",
+    "title",
+    "description",
+    "default",
+    "examples",
+    "required", # TODO: implement and remove from ignored list
```

Comment:
Decided to add "required" to the ignored key list for now. Currently the grammar is just a little bit over-zealous and considers every property of an object to be required, so this isn't a terrible failure mode. "Fixing" this felt beyond the scope of this PR, but I added a new issue to track this: https://github.com/guidance-ai/guidance/issues/906